overlays/emacs.nix

```
@@ -63,6 +63,9 @@ let
                         "${super.lib.getBin self.stdenv.cc.cc}/bin"
                         "${super.lib.getBin self.stdenv.cc.bintools}/bin"
                         "${super.lib.getBin self.stdenv.cc.bintools.bintools}/bin"
+                      ] ++ super.lib.optionals (self.stdenv.hostPlatform.isDarwin && super ? apple-sdk) [
+                        # The linker needs to know where to find libSystem on Darwin.
+                        "${super.apple-sdk.sdkroot}/usr/lib"
```

This is semantically more correct.

```suggestion
                      ] ++ super.lib.optionals (self.stdenv.hostPlatform.isDarwin && self ? apple-sdk) [
                        # The linker needs to know where to find libSystem on Darwin.
                        "${self.apple-sdk.sdkroot}/usr/lib"
```

(All those `super.lib` should be `self.lib` too, but that is out of the scope of this PR.)